astropy/nddata/ccddata.py

```diff
@@ -416,7 +416,11 @@ def _generate_wcs_and_update_header(hdr):
     wcs_header = wcs.to_header(relax=True)
     for k in wcs_header:
         if k not in _KEEP_THESE_KEYWORDS_IN_HEADER:
-            del new_hdr[k]
+            try:
+                del new_hdr[k]
+            except KeyError:
+                # Keyword does not exist in the original header
+                pass
```

Comment:
@astrofrog -- the goal was to avoid a header and WCS that conflict by keeping the WCS as a separate object and removing the WCS keywords from the header.

An alternative approach would be to make `CCDData.wcs` a property. That might be the better way to go, since we don't do anything to prevent the user from adding WCS to the header that is completely independent of whatever is stored in `.wcs`.